docs/source/developer/repomap.rst

```diff
@@ -0,0 +1,447 @@
+- ``numba/_npymath_exports.c`` - Export function pointer table to NumPy C math functions
+- ``numba/_dynfuncmod.c`` - Python extension module exporting _dynfunc.c functionality
+- ``numba/_dynfunc.c`` - C level Environment and Closure objects (keep in sync with numba/target/base.py)
+- ``numba/mathnames.h`` - Macros for defining names of math functions
+- ``numba/_pymodule.h`` - C macros for Python 2/3 portable naming of C API functions
+- ``numba/_math_c99.{h,c}`` - C99 math compatibility (needed Python 2.7 on Windows, compiled with VS2008)
+- ``numba/mviewbuf.c`` - Handles Python memorviews
+- ``numba/_typeof.{h,c}`` - C implementation of type fingerprinting, used by dispatcher
+- ``numba/_numba_common.h`` - Portable C macro for marking symbols that can be shared between object files, but not outside the library.
+
+
+
+Misc Support
+''''''''''''
+
+- ``numba/_version.py`` - Updated by versioneer
+- ``numba/runtime`` - Global singleton that manages memory allocated on the heap by Numba-compiled functions
+- ``numba/ir_utils.py`` - Utility functions for working with Numba IR data structures 
+- ``numba/cgutils.py`` - Utility functions for generating common code patterns in LLVM IR
+- ``numba/six.py`` - Vendored subset of ``six`` package for Python 2 + 3 compatibility
+- ``numba/io_support.py`` - Workaround for various names of StringIO in different Python versions (should this be in six?)
+- ``numba/utils.py`` - Python 2 backports of Python 3 functionality (also imports local copy of ``six``)
+- ``numba/appdirs.py`` - Vendored package for determining application config directories on every platform
+- ``numba/compiler_lock.py`` - Global compiler lock because our usage of LLVM is not thread-safe
+- ``numba/special.py`` - Python stub implementations of special Numba functions (prange, gdb*)
+- ``numba/servicelib`` - Should be removed?
+- ``numba/itanium_mangler.py`` - Python implementation of Itanium C++ name mangling
+- ``numba/findlib.py`` - Helper function for locating shared libraries on all platforms
+- ``numba/debuginfo.py`` - Helper functions to construct LLVM IR debug info
+- ``numba/unsafe`` - ``@intrinsic`` helper functions that can be used to implement direct memory/pointer manipulation from nopython mode functions
+- ``numba/unsafe/refcount.py`` - Read reference count of object
+- ``numba/unsafe/tuple.py`` - Replace a value in a tuple slot
+- ``numba/unsafe/ndarray.py`` - NumPy array helpers
+- ``numba/unsafe/bytes.py`` - Copying and dereferencing data from void pointers
+- ``numba/dummyarray.py`` - ???
+- ``numba/callwrapper.py`` - Handles argument unboxing and releasing the GIL when moving from Python to nopython mode
+- ``numba/ctypes_support.py`` - Import this instead of ``ctypes`` to workaround portability issue with Python 2.7
+- ``numba/cffi_support.py`` - Alias of numba.typing.cffi_utils for backward compatibility (still needed?)
+- ``numba/numpy_support.py`` - Helper functions for working with NumPy and dtypes
```

Comment:
also Numba type interoperability with NumPy.

Reply:
rephrased to make this clear